round_tripper.go

```diff
@@ -22,22 +23,20 @@ type Transport struct {
 	// if nil, emit error log by log package, and ignore it.
 	UnusualError func(err error) error
 
-	har   *HARContainer
-	mutex sync.Mutex
+	Container *HARContainer
 }
 
 func (h *Transport) init() {
-	if h.har != nil {
+	if h.Container != nil {
 		return
 	}
 
-	h.mutex.Lock()
-	defer h.mutex.Unlock()
-	if h.har != nil {
-		return
-	}
+	containerPtr := (*unsafe.Pointer)(unsafe.Pointer(&h.Container))
+	atomic.CompareAndSwapPointer(containerPtr, unsafe.Pointer(nil), unsafe.Pointer(NewHARContainer()))
```

Comment:
Since `HARContainer` is now shared, I needed to move the mutex into that struct.

I've opted to use an atomic compare and swap for the lazy initialization of the `Container` field on `Transport`, but I'm also happy to just make this a mutex as well if that reads clearer.

A third option is to use the new [`atomic.Pointer` type](https://pkg.go.dev/sync/atomic#Pointer) which offers a more ergonomic API for a compare-and-swap, but would necessitate bumping this library's minimum required Go version to 1.19.